Add Release Doc (#503)
bpfd-dev/bpfd: RELEASE.md

# Release Process

## Overview

A release for the bpfd project is comprised of the following major components:

> - Built bpfd + bpfctl binaries

Reviewer: ```suggestion
- bpfd + bpfctl binaries
```

Reviewer: "Built" seems redundant.

Author: Done

- Core GRPC API protobuf definitions
- Kubernetes Custom Resource Definitions (CRDs)
- Corresponding go pkg in the form of `github.com/bpfd-dev/bpfd` which includes the following:
  - `github.com/bpfd-dev/bpfd/clients/gobpfd/v1`: The go client for the bpfd GRPC API
  - `github.com/bpfd-dev/bpfd/bpfd-operator/apis`: The go bindings for the
    bpfd CRD API
  - `github.com/bpfd-dev/bpfd/bpfd-operator/pkg/client`: The autogenerated
    clientset for the bpfd CRD API
  - `github.com/bpfd-dev/bpfd/bpfd-operator/pkg/helpers`: The provided bpfd CRD
    API helpers.
- Corresponding `bpfd-api`, `bpfd`, and `bpfctl` rust crates which house the rust client for the bpfd GRPC API
- The following core component container images with tag <RELEASE_VERSION>:
  - `quay.io/bpfd/bpfd`
  - `quay.io/bpfd/bpfctl`
  - `quay.io/bpfd/bpfd-operator`
  - `quay.io/bpfd/bpfd-agent`
  - `quay.io/bpfd/bpfd-operator-bundle`

> - The relevant example bytecode container images with tag <RELEASE_VERSION> who's source
>   code is located in the bpfd project:

Reviewer: ```suggestion
- The relevant example bytecode container images with tag <RELEASE_VERSION> from source
  code located in the bpfd project:
```

  - `quay.io/bpfd-bytecode/go_xdp_counter`
  - `quay.io/bpfd-bytecode/go_tc_counter`
  - `quay.io/bpfd-bytecode/go_tracepoint_counter`
- The relevant example userspace container images with tag <RELEASE_VERSION> from source

>   code is located in the bpfd project:

Author: Done :)

> - The relevant example userspace container images with tag <RELEASE_VERSION> who's source
>   code is located in the bpfd project:

Reviewer: ```suggestion
- The relevant example userspace container images with tag <RELEASE_VERSION> from source
  code located in the bpfd project:
```
